src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,162 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size = 0;
+    private Node<T> first;
+    private Node<T> last;
+
+    public MyLinkedList() {
+        first = new Node<>(null, null, null);
+        last = new Node<>(null, null, null);
+    }
+
+    private void checkIndexException(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException();
+        }
+    }
+
+    private void addToTail(T value) {
+        Node<T> newElem = new Node<>(last, value, null);
+        if (last.item == null) {
```

Comment:
@Korfat this is a bit confusing: you are checking the last.item, but using first.item... The other developer should find the code and move to it